src/Parser/Tokenizer/Tokenizer.php

```diff
@@ -162,40 +163,40 @@ public static function templateLiteral(\Iterator $fragments): \Iterator
 
         $fragments->next();
 
-        $escape = false;
-
         while ($fragments->valid()) {
-            /** @var Fragment $fragment */
-            $fragment = $fragments->current();
 
-            if (!$escape && $fragment->value === '`') {
-                yield from $buffer->flush(TokenType::STRING_QUOTED);
-                $buffer->append($fragments->current());
-                yield from $buffer->flush(TokenType::TEMPLATE_LITERAL_END);
-                $fragments->next();
-                return;
-            }
+            switch ($fragments->current()->value) {
+                case '`':
+                    yield from $buffer->flush(TokenType::STRING_QUOTED);
+                    $buffer->append($fragments->current());
+                    yield from $buffer->flush(TokenType::TEMPLATE_LITERAL_END);
+                    $fragments->next();
+                    return;
 
-            if (!$escape && $fragment->value === '$') {
-                $dollarSignBuffer = Buffer::empty()->append($fragments->current());
-                $fragments->next();
-                $nextFragment = $fragments->current();
+                case '$':
+                    $dollarSignBuffer = Buffer::empty()->append($fragments->current());
+                    $fragments->next();
+                    $nextFragment = $fragments->current();
```

Comment:
This breaks, unless `$fragments->valid()` is checked first. (Looks like I broke this one first :sweat_smile:)